test/roby_app/remote_processes/test_remote_processes.rb

```diff
@@ -128,20 +127,124 @@
             root_loader.register_deployment_model(deployment)
             process = client.start(
                 "syskit_tests_empty", "syskit_tests_empty", {},
-                wait: true, oro_logfile: nil, output: "/dev/null"
+                wait: false, oro_logfile: nil, output: "/dev/null"
             )
             assert_same deployment, process.model
         end
     end
 
+    describe "waits for the process to be running" do
+        before do
+            start_and_connect_to_server
+        end
+
+        it "returns a hash with information about a process and its tasks" do
+            client.start(
+                "syskit_tests_empty", "syskit_tests_empty",
+                { "syskit_tests_empty" => "syskit_tests_empty" },
+                wait: false, oro_logfile: nil, output: "/dev/null"
+            )
+            result = nil
+            loop do
+                result = client.wait_running("syskit_tests_empty")
+                break if result["syskit_tests_empty"]&.key?(:iors)
+            end
+            assert_includes(result, "syskit_tests_empty")
+            assert_includes(result["syskit_tests_empty"], :iors)
+            assert_includes(result["syskit_tests_empty"][:iors], "syskit_tests_empty")
+        end
+
+        it "returns a hash without any info when the process didnt get its tasks ior" do
+            client.start(
+                "syskit_tests_empty", "syskit_tests_empty",
+                { "syskit_tests_empty" => "syskit_tests_empty" },
+                wait: false, oro_logfile: nil, output: "/dev/null"
+            )
+            result = client.wait_running("syskit_tests_empty")
+            assert_equal({ "syskit_tests_empty" => nil }, result)
+        end
+
+        it "reports a Orocos::NotFound error specific to a process" do
+            not_found_error_message = "syskit_tests_empty was started but crashed"
+            orocos_proc_mock =
+                flexmock(Orocos::ProcessBase).new_instances
+                                             .should_receive(:wait_running)
+            orocos_proc_mock.and_raise(Orocos::NotFound, not_found_error_message)
+                            .ordered
```

Comment:
do everything in one block

~~~
flexmock(...)
  .new_instances
  .should_receive(:wait_running)
  .and_raise(....)
  .once
~~~